### PR TITLE
bp: Add WARN Logging on Slow Transport Message Handling

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
https://github.com/elastic/elasticsearch/commit/5112c173194b70d2ed17f2ed3071dc0c9a44b76e

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/transport/InboundHandler.java [hunk 1]
```diff
@@ -76,64 +79,78 @@ void setMessageListener(TransportMessageListener listener) {
...
+                assert message.isShortCircuit() == false;
+                final TransportResponseHandler<?> handler;
+                long requestId = header.getRequestId();
+                if (header.isHandshake()) {
                     handler = handshaker.removeHandlerForHandshake(requestId);
                 } else {
-                    handler = theHandler;
-                }
-            }
-            // ignore if its null, the service logs it
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/org/elasticsearch/transport/InboundHandler.java:113]: git diff is not really clear, what I actually did is wrapping the whole method in a try block and added finally to align with linked patch https://github.com/elastic/elasticsearch/commit/5112c173194b70d2ed17f2ed3071dc0c9a44b76e#diff-e7af60ad8aea4d9c61ee4ae02164f46abee2943b8bc93c88161a9f03276c2cbfR149-R155

File: server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java [hunk 2]
```diff
@@ -256,6 +255,43 @@ public void testClosesChannelOnErrorInHandshakeWithIncompatibleVersion() throws
...
+            final InboundMessage requestMessage =
+                new InboundMessage(requestHeader, ReleasableBytesReference.wrap(BytesArray.EMPTY), () -> {
+                    try {
+                        TimeUnit.SECONDS.sleep(1L);
+                    } catch (InterruptedException e) {
+                        throw new AssertionError(e);
+                    }
+                });
+            requestHeader.actionName = TransportHandshaker.HANDSHAKE_ACTION_NAME;
+            requestHeader.bwcNeedsToReadVariableHeader = false;
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java:285]: this is reflecting headers provision in https://github.com/elastic/elasticsearch/commit/5112c173194b70d2ed17f2ed3071dc0c9a44b76e#diff-b5518bc3e3b5862afbc9f1978088945d8c71427f421e9a703e085931ffcac577R272

We don't have `headers` Set, we have a flag.
See https://github.com/crate/crate/commit/8fbec5d8d386e5804080356020cb573cadbac678#diff-0eef2442af29607a3a1ed7df397bd704179d41d84fc0a135fe96b814513a5464R88 and 
https://github.com/elastic/elasticsearch/commit/2ccddbfa88e#diff-0eef2442af29607a3a1ed7df397bd704179d41d84fc0a135fe96b814513a5464R95

Comment (matriv) [reply server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java:285]: Maybe add a comment on top of the flag set to explain?

**Review by matriv — COMMENTED** [review]
I've added a comment for the test and the bwc headers flag.
Imho, it's not necessary to document, it's a flag affecting logging and not something a user would need to manipulate under normal circumstances. But please wait to hear somebody else opinion as well, maybe @mfussenegger ?

File: server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java (hunk 2 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java:287]: Added comment in https://github.com/crate/crate/pull/13409/commits/29a3516b5d1397a05fca7c4b4ed2cd9a186e4b00

**Review by matriv — APPROVED** [review]
Thank you, let's wait for other's opinion on docs as discussed.

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**